src/stream.rs

```diff
@@ -1182,8 +1274,10 @@ impl RangeBuf {
     /// Creates a new `RangeBuf` from the given slice.
     pub(crate) fn from(buf: &[u8], off: u64, fin: bool) -> RangeBuf {
         RangeBuf {
-            data: Vec::from(buf),
+            data: Arc::new(Vec::from(buf)),
```

Comment:
Just curious why `Arc` is needed now.

Reply:
Arc usage means there is only one underlying data store copy. 

When the split method is called later and `data` is cloned, it only increments the ref count. The new RangeBuf just has a different view onto the underlying data.

When all RangBufs for this data get tidied up, the ref count hits zero and the memory freed at the right time automatically.